MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -419,7 +419,7 @@
     // macOS before 10.15.5 cannot use shared memory for texel buffers.
     usesTexelBuffer = usesTexelBuffer && (mtlFeats.sharedLinearTextures || !isMemoryHostCoherent());
 
-    if (_image->_isLinearForAtomics || (usesTexelBuffer && mtlFeats.placementHeaps)) {
+    if ((_deviceMemory && _image->_isLinearForAtomics) || (usesTexelBuffer && mtlFeats.placementHeaps)) {
```

Comment:
I apologize for the extra churn, but it occurs to me that the entire `if`-`else` chain ought to be guarded by `if (_deviceMemory)`. Then the null check on line 428 could go away.

Reply:
Not a problem at all. I had originally put together a change that bailed out early and decided it would be safer to do something more targeted; but because `_deviceMemory` effectively gets checked in so many places, the entire function ends up being a no-op after the `MVKResource::bindDeviceMemory` call.